manubot/process/tests/test_ci.py

```diff
@@ -9,50 +9,92 @@
-    assert info['repo_name'] == 'manubot'
-    assert info['commit']
-    assert info['triggering_commit']
-    assert info['build_url'].startswith('https://travis-ci.com/manubot/manubot/builds/')
-    assert info['job_url'].startswith('https://travis-ci.com/manubot/manubot/jobs/')
-    # test add_manuscript_urls_to_ci_params
-    info_updated = add_manuscript_urls_to_ci_params(info)
-    assert info is info_updated
-    assert re.fullmatch(
-        pattern=r"https://manubot\.github\.io/manubot/v/[0-9a-f]{40}/",
-        string=info['manuscript_url'],
-    )
+@pytest.fixture
+def info():
+    return get_continuous_integration_parameters()
 
 
-@pytest.mark.skipif('APPVEYOR' not in os.environ, reason='tests environment variables set by AppVeyor builds only')
-def test_get_continuous_integration_parameters_appveyor():
-    info = get_continuous_integration_parameters()
-    assert info is not None
-    assert info['provider'] == 'appveyor'
-    assert info['provider_account'] == 'manubot'
-    assert info['repo_slug'] == 'manubot/manubot'
-    assert info['repo_owner'] == 'manubot'
-    assert info['repo_name'] == 'manubot'
-    assert info['commit']
-    assert info['triggering_commit']
-    assert info['build_url'].startswith('https://ci.appveyor.com/project/manubot/manubot/builds/')
-    assert info['job_url'].startswith('https://ci.appveyor.com/project/manubot/manubot/build/job/')
-    # test add_manuscript_urls_to_ci_params
-    info_updated = add_manuscript_urls_to_ci_params(info)
-    assert info is info_updated
-    assert re.fullmatch(
-        pattern=r"https://ci\.appveyor\.com/project/manubot/manubot/builds/[0-9]+/artifacts",
-        string=info['manuscript_url'],
-    )
+@pytest.mark.skipif(
+    'CI' not in os.environ,
+    reason='We are not on Travis or Appveyor.')
+class Test_get_continuous_integration_parameters():
+
+    def test_not_empty(self, info):
+        assert info is not None
+
+    def test_repo_name(self, info):
+        """
+        We assume repo name stays the same in repo forks.
+        """
+        assert info['repo_name'] == 'manubot'
+        assert info['repo_slug'].endswith('manubot')
+
+    def test_repo_slug(self, info):
+        # APPVEYOR_PROJECT_NAME - project name
+        # APPVEYOR_PROJECT_SLUG - project slug (as seen in project details URL)
+        # https://www.appveyor.com/docs/environment-variables/
+        #
+        # TRAVIS_REPO_SLUG: The slug (in form: owner_name/repo_name) of the repository currently being built.
+        # https://docs.travis-ci.com/user/environment-variables/
+        if 'TRAVIS' in os.environ:
+            assert info['repo_slug'] == os.environ['TRAVIS_REPO_SLUG']
+        elif 'APPVEYOR' in os.environ:
+            assert info['repo_slug'] == os.environ['APPVEYOR_PROJECT_SLUG']
+        else:
+            assert False
+
+    def test_commits_are_not_empty(self, info):
+        assert info['commit']
+        assert info['triggering_commit']
```

Comment:
Afraid this will fail on unsupported CI platforms like in #136. Anyway to make this dependent on a supported CI platform having been detected?

Reply:
This is similar to code already used, or I'm missing something?

Reply:
linked to build systems with `@pytest.mark.skipif('TRAVIS' not in os.environ or 'APPVEYOR' not in os.environ, reason=...)`